apps/astarte_appengine_api/lib/astarte_appengine_api/device/queries.ex

```diff
@@ -847,6 +852,87 @@ defmodule Astarte.AppEngine.API.Device.Queries do
     end
   end
 
+  def insert_metadata(client, device_id, metadata_key, metadata_value) do
+    insert_metadata_statement = """
+    UPDATE devices
+    SET metadata[:metadata_key] = :metadata_value
+    WHERE device_id = :device_id
+    """
+
+    query =
+      DatabaseQuery.new()
+      |> DatabaseQuery.statement(insert_metadata_statement)
+      |> DatabaseQuery.put(:metadata_key, metadata_key)
+      |> DatabaseQuery.put(:metadata_value, metadata_value)
+      |> DatabaseQuery.put(:device_id, device_id)
+      |> DatabaseQuery.consistency(:each_quorum)
```

Comment:
If this is `each_quorum`, also the DELETE query should be `:each_quorum`. What do you think about these consistency levels @bettio?